### ISSUE TITLE
Unable to build using new versioning.

### ISSUE THREAD

**Issue by ianwallen** [metadata101/iso19139.ca.HNAP#189]
Since changes from 
https://github.com/geonetwork/core-geonetwork/pull/5096

I cannot get the build to work correctly.

In our case we are trying to do the following
> * ⛔  GeoNetwork `3.10.6-0`, schemas/pom.xml `3.10.6-0`, schema plugin `3.10-SNAPSHOT`

I'm trying to use the latest HNAP within the 3.10.6 release  
I applied the commits from PR https://github.com/geonetwork/core-geonetwork/pull/5096
(I did replace all 3.10-SNAPSHOT back to 3.10.6-0).

I keep getting the following error.
```
[ERROR]   The project org.geonetwork-opensource.schemas:schema-iso19139.ca.HNAP:3.10-SNAPSHOT (C:\Projects\TADAP\geonetwork-dev\schemas\iso19139.ca.HNAP\pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for org.geonetwork-opensource.schemas:schema-iso19139.ca.HNAP:3.10-SNAPSHOT: Could not find artifact org.geonetwork-opensource.schemas:schemas:pom:3.10-SNAPSHOT and 'parent.relativePath' points at wrong local POM @ line 5, column 11 -> [Help 2]
```

**Comment by jodygarnett** [comment]
The parent pom changed from `org.geonetwork-opensource` to `org.geonetwork-opensource.schemas` I think.

When you return to `3.10.6` you may need to update the [pom.xml](https://github.com/metadata101/iso19139.ca.HNAP/blob/3.10.x/pom.xml) both?

```
  <parent>
    <artifactId>schemas</artifactId>
    <groupId>org.geonetwork-opensource</groupId>
    <version>3.10.6-0</version>
  </parent>
```

Let me double check: 

[3.10.6-0](https://github.com/geonetwork/core-geonetwork/blob/3.10.6/schemas/pom.xml#L36-L39) has:
```
  <modelVersion>4.0.0</modelVersion>
  <artifactId>schemas</artifactId>
  <name>GeoNetwork schema plugins</name>
  <packaging>pom</packaging>
```

[3.10.x](https://github.com/geonetwork/core-geonetwork/blob/3.10.x/schemas/pom.xml#L34-L38) has:
```
  <modelVersion>4.0.0</modelVersion>
  <groupId>org.geonetwork-opensource.schemas</groupId>
  <artifactId>schemas</artifactId>
```

**Comment by davidblasby** [comment]
Here is my resulting HNAP pom.xml

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xmlns="http://maven.apache.org/POM/4.0.0"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <parent>
    <artifactId>schemas</artifactId>
    <groupId>org.geonetwork-opensource.schemas</groupId>
    <version>3.10-SNAPSHOT</version>
    <!-- tested with 3.10.6-SNAPSHOT -->
  </parent>
  <modelVersion>4.0.0</modelVersion>

  <artifactId>schema-iso19139.ca.HNAP</artifactId>
  <name>Schema Plugin HNAP ISO19139/119</name>
  <version>3.10-SNAPSHOT</version>
  <description>
     GeoNetwork schema plugin for HNAP ISO19139/119 standard
  </description>

  <dependencies>
    <dependency>
      <groupId>org.geonetwork-opensource.schemas</groupId>
      <artifactId>schema-core</artifactId>
      <version>3.10-SNAPSHOT</version>
      <scope>provided</scope>
    </dependency>
    <dependency>
     <groupId>org.geonetwork-opensource</groupId>
     <artifactId>core</artifactId>
     <version>${gn.project.version}</version>
     <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>org.xmlunit</groupId>
      <artifactId>xmlunit-core</artifactId>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>org.xmlunit</groupId>
      <artifactId>xmlunit-matchers</artifactId>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>log4j</groupId>
      <artifactId>log4j</artifactId>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>org.geonetwork-opensource</groupId>
      <artifactId>common</artifactId>
      <version>${gn.project.version}</version>
      <scope>provided</scope>
    </dependency>
    <!-- test scope -->
    <dependency>
      <groupId>${project.groupId}</groupId>
      <artifactId>schema-core</artifactId>
      <version>3.10-SNAPSHOT</version>
      <type>test-jar</type>
      <scope>test</scope>
    </dependency>
  </dependencies>

  <build>
    <resources>
      <!-- packaged into jar -->
      <resource>
        <directory>src/main/config/codelist/local/thesauri/theme</directory>
        <targetPath>thesarus/theme</targetPath>
      </resource>
      <resource>
        <directory>src/main/config/codelist/local/thesauri/place</directory>
        <targetPath>thesarus/place</targetPath>
      </resource>
      <resource>
        <directory>src/main/config/conversion</directory>
        <targetPath>xsl/conversion</targetPath>
      </resource>
      <resource>
        <directory>src/main/config/translations</directory>
        <targetPath>META-INF/catalog/locales</targetPath>
      </resource>
      <resource>
        <directory>src/main/resources</directory>
      </resource>
      <!-- packaged into zip -->
      <resource>
        <directory>src/main/plugin/iso19139.ca.HNAP</directory>
      </resource>
    </resources>
    <testResources>
      <testResource>
        <directory>src/test/resources/org/fao/geonet</directory>
      </testResource>
    </testResources>

    <plugins>
      <plugin>
        <artifactId>maven-jar-plugin</artifactId>
        <configuration>
          <includes>
            <include>ca/**</include>
            <include>META-INF/**</include>
            <include>thesarus/**</include>
            <include>xsl/**</include>
            <include>config-spring-geonetwork.xml</include>
            <include>rebel.xml</include>
          </includes>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-assembly-plugin</artifactId>
        <executions>
          <execution>
            <id>plugin-assembly</id>
            <phase>package</phase>
            <goals><goal>single</goal></goals>
            <inherited>false</inherited>
            <configuration>
             <appendAssemblyId>false</appendAssemblyId>
             <descriptors>
              <descriptor>src/assembly/schema-plugin.xml</descriptor>
             </descriptors>
            </configuration>
          </execution>
        </executions>
      </plugin>
    </plugins>
  </build>

  <profiles>
    <profile>
      <id>run-static-analysis</id>
      <activation>
        <property>
          <name>!skipTests</name>
        </property>
      </activation>
      <build>
        <plugins>
          <plugin>
            <groupId>org.codehaus.mojo</groupId>
            <artifactId>findbugs-maven-plugin</artifactId>
          </plugin>
        </plugins>
      </build>
    </profile>
    <profile>
      <id>docs</id>
      <build>
        <plugins>
          <plugin>
            <artifactId>maven-antrun-plugin</artifactId>
            <executions>
              <execution>
                <phase>compile</phase>
                <configuration>
<target>
   <property name="sphinx_build_options"
             value="-q -N -W --keep-going -n -j auto"/>
   <property name="theme"
             value="sphinx_rtd_theme"/>
   <property name="doctrees"
             value="target/doctrees"/>
   <property name="outputdir"
             value="target/html"/>
   <property name="srcdir"
             value="src/sphinx"/>

   <mkdir dir="${outputdir}"/>
   <exec executable="sphinx-build" failonerror="true">
     <arg line="${sphinx_build_options}"/>
     <arg line="-b html"/>
     <arg line="-Dhtml_theme=${theme}"/>
     <arg line="-d ${doctrees}"/>
     <arg path="${srcdir}"/>
     <arg path="${outputdir}"/>
   </exec>
</target>
                </configuration>
                <goals><goal>run</goal></goals>
              </execution>
            </executions>
          </plugin>
          <plugin>
            <artifactId>maven-assembly-plugin</artifactId>
            <executions>
              <execution>
                <id>docs-assembly</id>
                <phase>package</phase>
                <goals><goal>single</goal></goals>
                <inherited>false</inherited>
                <configuration>
                 <appendAssemblyId>false</appendAssemblyId>
                 <descriptors>
                  <descriptor>src/assembly/docs.xml</descriptor>
                 </descriptors>
                 <finalName>${project.build.finalName}-doc</finalName>
                </configuration>
              </execution>
            </executions>
          </plugin>
          <plugin>
            <artifactId>maven-resources-plugin</artifactId>
            <executions>
              <execution>
                <id>publish</id>
                <phase>install</phase>
                <goals><goal>copy-resources</goal></goals>
                <configuration>
                  <outputDirectory>docs</outputDirectory>
                  <resources>
                    <resource>
                      <directory>${project.build.directory}/html</directory>
                    </resource>
                  </resources>
                  <addDefaultExcludes/>
                  <encoding>UTF-8</encoding>
                  <overwrite>true</overwrite>
                </configuration>
               </execution>
             </executions>
           </plugin>
        </plugins>
      </build>
    </profile>
  </profiles>
</project>
```

This looks to be working.

Here's the diff:

**% git diff pom.xml**
```
diff --git a/pom.xml b/pom.xml
index 702ce30..2b56656 100644
--- a/pom.xml
+++ b/pom.xml
@@ -4,28 +4,28 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <parent>
     <artifactId>schemas</artifactId>
-    <groupId>org.geonetwork-opensource</groupId>
-    <version>3.7</version>
+    <groupId>org.geonetwork-opensource.schemas</groupId>
+    <version>3.10-SNAPSHOT</version>
     <!-- tested with 3.10.6-SNAPSHOT -->
   </parent>
   <modelVersion>4.0.0</modelVersion>
 
   <artifactId>schema-iso19139.ca.HNAP</artifactId>
   <name>Schema Plugin HNAP ISO19139/119</name>
-  <version>3.10.6-0</version>
+  <version>3.10-SNAPSHOT</version>
   <description>
      GeoNetwork schema plugin for HNAP ISO19139/119 standard
   </description>
 
   <dependencies>
     <dependency>
-      <groupId>${project.groupId}</groupId>
+      <groupId>org.geonetwork-opensource.schemas</groupId>
       <artifactId>schema-core</artifactId>
-      <version>3.7</version>
+      <version>3.10-SNAPSHOT</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
-     <groupId>${project.groupId}</groupId>
+     <groupId>org.geonetwork-opensource</groupId>
      <artifactId>core</artifactId>
      <version>${gn.project.version}</version>
      <scope>provided</scope>
@@ -46,7 +46,7 @@
       <scope>provided</scope>
     </dependency>
     <dependency>
-      <groupId>${project.groupId}</groupId>
+      <groupId>org.geonetwork-opensource</groupId>
       <artifactId>common</artifactId>
       <version>${gn.project.version}</version>
       <scope>provided</scope>
@@ -55,7 +55,7 @@
     <dependency>
       <groupId>${project.groupId}</groupId>
       <artifactId>schema-core</artifactId>
-      <version>3.7</version>
+      <version>3.10-SNAPSHOT</version>
       <type>test-jar</type>
       <scope>test</scope>
     </dependency>
```

**Comment by ianwallen** [comment]
I don't think you understood the problem.

We were previously building the code by pointing directly to the github submodule and pulling the code directly in as a submodule without any changes.  We did not have to change any code so we did not have to do any fork.

All your suggested require us to fork the HNAP plugin to make the custom changes in order to make it work.  I was trying to avoid doing that and I thought that as a project, we were trying to avoid making people fork the code just to do minor configuration.   This seems to be a step backwards.

Unfortunately, we now have to fork the code due to these recent changes.
